enhancements/olm-deprecate-copied-csvs.md

```diff
@@ -0,0 +1,155 @@
+
+This enhancement details the steps required to deprecate and replace OLM's [Copied CSV](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/#copied-csvs) feature in higher fidelity than [ever before](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#deprecate-copied-clusterserviceversions). Importantly, it introduces a feature toggle that disables/enables CSV copying for all of OLM.
+
+## Motivation
+
+When an Operator is installed by OLM, a stripped down copy -- e.g. a stub -- of its [CSV](https://olm.operatorframework.io/docs/concepts/crds/clusterserviceversion/) is created in every namespace the Operator is configured to watch (see [OperatorGroup](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/); collectively, these are known as Copied CSVs. This means that the number of Copied CSVs on a given cluster grows proportionally with both the number of namespaces and installed Operators. In the worst case, a cluster can have a Copied CSV for every Operator in every namespace, that is `|copied| = |operators|*|namespaces|`. For especially large clusters, with namespaces and installed operators tending in the hundreds or thousands, Copied CSVs consume an untenable amount of resources; e.g. OLM's memory usage, cluster Etcd limits, networking, etc. For these reasons, it's imperative that Copied CSVs be replaced with a more resource friendly successor.
+
+### Goals
+
+- Allieviate the resource utilization incurred by Copied CSVs
+- Highlight features that subsume the featureset of Copied CSVs
+- Enumerate the steps necessary to deprecate and remove Copied CSVs from use
+
+### Non-Goals
+
+- Provide a zero-effort migration to clients
+
+## Proposal
+
+For each non-Copied CSV, a single [Operator](https://github.com/operator-framework/enhancements/blob/master/enhancements/simplify-olm-apis.md#proposal) resource will be generated. This cluster-scoped resource will take the place of any related Copied CSVs.
+
+### User Stories
+
+#### Feature Toggle
+
+As a Cluster Admin, I want a way to disable Copied CSVs so that I can reduce resource consumption ahead of the feature's deprecation.
+
+#### View "Available" Operators
+
+As a User, I want a way to view the operators that can act in a given namespace I have access to.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Operator Cardinality
+
+Today, Operator resources are generated on a per-Subscription basis. Since CSVs -- and thus Copied CSVs -- can exist without matching Subscriptions, OLM's controllers will be extended to generate additional Operator resources for these "free CSVs" such that:
+
+- there exists at least one`Operator` resource for every CSV
+- there exists at least one `Operator` resource for every `Subscription`
+
+That is, all CSVs installed by a given `Subscription` share an `Operator` resource __and__ any CSVs that weren't installed by a `Subscription` are related to their own `Operator` resource.
```

Comment:
Is it worth mentioning the case where a CSV was installed via a subscription and then the subscription was removed? The Operator related to the CSV would still be around, but maybe its properties change since there is no underlying subscription?

Reply:
I had a somewhat adjacent question to this: how are we correlating which CSVs or Subscriptions are represented by an Operator resource? Does the Operator resource track references in it's status sub-resource, or that information is tracked as an annotation, etc.?